lib/extensions/ar_visible_attribute.rb

```diff
@@ -0,0 +1,23 @@
+module ArVisibleAttribute
+  extend ActiveSupport::Concern
+
+  included do
+    cattr_accessor :hidden_attribute_names
```

Comment:
`cattr_accessor`, when changed in a subclass, also changes the value of the parent attribute and is shared to every other subclass. Since ems sets `aggregate_memory`, and then for every active record class, that attribute will be present on every active record model.

Reply:
cattr_accessor uses class variables (`@@`) under the covers, and I agree it will lead to bugs.  Prefer `class_attribute` which uses class instance (`@`) variables under the covers and is more in line with expectations.

Reply:
[this is just a note for me]

using `class_attribute` means we will need to take into account this:
```
# Appending in child changes both parent and child because it is the same object:
Subclass.setting << :foo
Base.setting               # => [:foo]
Subclass.setting           # => [:foo]

# Use setters to not propagate changes:
Base.setting = []
Subclass.setting += [:foo]
```

Reply:
THAT makes it more clear.

also:

```ruby
settings += [:foo]
# same as
var settings = @@settings + [:foo]
```

yes I used `var` from javascript and used `@@` vs `class.@` but he meaning is pretty clear and the pitfall too

Reply:
added a test to protect us from this